scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/AvroSortedBucketIO.java

```diff
@@ -62,55 +61,116 @@ public static <T extends SpecificRecordBase> Read<T> read(
   }
 
   /** Returns a new {@link Write} for Avro generic records. */
-  public static <K> Write<K, GenericRecord> write(
-      Class<K> keyClass, String keyField, Schema schema) {
-    return AvroSortedBucketIO.newBuilder(keyClass, keyField).setSchema(schema).build();
+  public static <K1> Write<K1, Void, GenericRecord> write(
+      Class<K1> keyClassPrimary, String keyFieldPrimary, Schema schema) {
+    return AvroSortedBucketIO.<K1, Void, GenericRecord>newBuilder(
+            keyClassPrimary, null, keyFieldPrimary, null)
```

Comment:
As mentioned above, changing the ordering would look more natural
```suggestion
    return AvroSortedBucketIO.<K1, Void, GenericRecord>newBuilder(
            primaryKeyClass, primaryKeyField, null, null)
```